### PR TITLE
[MI-2885]: Fix switch component

### PR DESCRIPTION
**Pull request by SaurabhSharma-884**

### CODE REVIEW

File: src/components/Switch/Switch.component.tsx [hunk 1]
```diff
@@ -22,25 +22,30 @@ import {StyledSwitch} from './Switch.styles';
  *
  */
 export const Switch = (props: SwitchProps) => {
-    const {id, label, subLabel, error, className = '', fullWidth = false, ...restProps} = props;
+    const {id, label, subLabel, error, className = '', fullWidth = false, disabled, size = 'md', ...restProps} = props;
```
Comment (Kshitij-Katiyar) [inline src/components/Switch/Switch.component.tsx:25]: Is there any need for constants for md sm etc. One use case could be that we can give detailed name so for others it would be easier to understand these sizes and not have to find the comment in the switch file.

**Review by Kshitij-Katiyar — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -72,10 +73,10 @@
 		"lodash": "4.17.21",
 		"postcss": "8.4.16",
 		"postcss-cli": "10.0.0",
-		"postcss-loader": "^4.3.0",
+		"postcss-loader": "4.3.0",
 		"prettier": "2.3.2",
 		"react": "16.13.1",
-		"react-bootstrap": "1.6.1",
+		"react-bootstrap": "github:mattermost/react-bootstrap#c6957962364e0818a51bbfd13e89919903b422d6",
```
Comment (avas27JTG) [inline package.json:79]: why this change?

File: src/components/Switch/Switch.styles.ts [hunk 3]
```diff
@@ -1,120 +1,155 @@
...
 
-        '&:active': {
-            backgroundColor: colors.buttonPrimaryActive,
-            filter: 'none',
         },
 
-        '&:focus': {
-            borderColor: colors.primary_32,
-            boxShadow: 'none',
+        // Style for switch when disabled
```
Comment (avas27JTG) [inline src/components/Switch/Switch.styles.ts:121]:
```suggestion
        // Style for disabled switch
```

File: src/components/Switch/Switch.styles.ts [hunk 4]
```diff
@@ -1,120 +1,155 @@
...
-    '& input.form-check-input.is-invalid': {
-        borderColor: colors.error,
+        '& .mm-switch-input[disabled].mm-switch-input:checked': {
+            opacity: '32%',
+        },
 
-        '&:checked': {
-            borderColor: 'transparent',
-            backgroundColor: colors.error,
+        // Style for switch when checked
```
Comment (avas27JTG) [inline src/components/Switch/Switch.styles.ts:132]:
```suggestion
        // Style for checked switch
```

File: package.json (hunk 2 above)
Comment (SaurabhSharma-884) [reply package.json]: This change was visible because this branch was not updated. Fixed Now.

**Review by avas27JTG — APPROVED** [review]
